### ISSUE TITLE
Should we remove the encoders import from the @service macro?

### ISSUE THREAD

**Issue by fedefernandez** [frees-io/freestyle-rpc#369]
Base on the comment on https://github.com/frees-io/freestyle-rpc/issues/269#issuecomment-405244125 I'm raising the question here:

Should we remove the encoders import from the @service macro?

Cons:
* Users need to manually add the decoders import

Pros:
* Users can specify it's own encoders/decoders for fields inside products
* We can provide extra encoders/decoders like in #269

**Comment by juanpedromoreno** [comment]
The code generation would add those imports anyway at the top of the file, correct? How do users could change them in that case?

**Comment by fedefernandez** [comment]
I didn't think in the code generation. I guess we would need to add a new statement like (just a draft):
```
import scala "import _root_.freestyle.rpc.internal.encoders.pbd._";
```

**Comment by juanpedromoreno** [comment]
Yeah, but this would prevent to users to add their custom encoders/decoders, wouldn't it?

**Comment by fedefernandez** [comment]
Probably I didn't explain by my self right. I'd present a couple of cases defined with avdl files.

*Default encoders*
```
@namespace("io.frees.samples.protocol")
protocol SampleService {

  import idl "Model.avdl";

  io.frees.samples.protocol.SampleResponse  getResponse(io.frees.samples.protocol.SampleRequest request);
}
```

This will generate something like:
```scala
package io.frees.samples.protocol

import freestyle.rpc.protocol._
import freestyle.rpc.internal.encoders.avro._

@service trait SampleService[F[_]] {

  @rpc(Avro)
  def getResponse(request: io.frees.samples.protocol.SampleRequest): F[io.frees.samples.protocol.SampleResponse]
}
```

*Custom encoders*
```
@namespace("io.frees.samples.protocol")
protocol SampleService {

  import scala "com.sample.myEncoders._";
  import idl "Model.avdl";

  io.frees.samples.protocol.SampleResponse  getResponse(io.frees.samples.protocol.SampleRequest request);
}
```

This will generate something like:
```scala
package io.frees.samples.protocol

import freestyle.rpc.protocol._
import com.sample.myEncoders._

@service trait SampleService[F[_]] {

  @rpc(Avro)
  def getResponse(request: io.frees.samples.protocol.SampleRequest): F[io.frees.samples.protocol.SampleResponse]
}
```

Does it make sense? It'll allow us to manually specify several custom imports, not only the decoders one.

**Comment by juanpedromoreno** [comment]
I see, thanks for the clarification. However, that'd be out of the avro specs, right?

**Comment by fedefernandez** [comment]
Yes. If we want to fulfill the avro specs I think the only option is to use comments, but that would be a bit odd:
```
@namespace("io.frees.samples.protocol")
protocol SampleService {

  //scala "import com.sample.myEncoders._"
  import idl "Model.avdl";

  io.frees.samples.protocol.SampleResponse  getResponse(io.frees.samples.protocol.SampleRequest request);
}
```

**Comment by juanpedromoreno** [comment]
Another question I have: how could we enforce that server and client are using the same encoders/decoders to allow the communication?

**Comment by fedefernandez** [comment]
Sorry for the noise. It seems that the root issue was a shadowing problem. I've fixed #341 and I'm preparing a new PR for making the `BigDecimal` and java.time encoders/decoders optional.
